src/Core/Product/ProductInterface.php

```diff
@@ -0,0 +1,36 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Product;
+
+interface ProductInterface
+{
+    const REDIRECT_CATEGORY_301 = '301-category';
+    const REDIRECT_CATEGORY_302 = '302-category';
+    const REDIRECT_PRODUCT_301 = '301-product';
+    const REDIRECT_PRODUCT_302 = '302-product';
+    const REDIRECT_NOT_FOUND = '404';
```

Comment:
If you have a `REDIRECT_NOT_FOUND` for 404 you must be consistent and have `REDIRECT_PRODUCT_MOVED_PERMANENTLY` for 301 and `REDIRECT_PRODUCT_FOUND` for 302 :)

Reply:
not sure ProductInterface should store something about Http status code... also what do we expect from something like "301-product" ?

Reply:
In fact, they are not http status code, but just some key stored in database.

Reply:
@mickaelandrieu FYI https://github.com/PrestaShop/PrestaShop/blob/5cd74cb6e36501436a0ef437d048dcf250cb8b0d/controllers/front/ProductController.php#L134

Reply:
I am ok with the consistency problem, however I hesitate with REDIRECT_PRODUCT_MOVED_PERMANENTLY name
It is really clear for sure, but also very long which does not help the code readability that's why I used 301 as a suffix which is understandable right away
Would you be ok with using the same pattern for the 404 => REDIRECT_404 ?

Reply:
hum... not really fan to have number in const. 
What do you prefer @mickaelandrieu:

 * REDIRECT_CATEGORY_301
 * REDIRECT_CATEGORY_302
 * REDIRECT_PRODUCT_301
 * REDIRECT_PRODUCT_302 
 * REDIRECT_404

or 

* REDIRECT_TYPE_NOT_FOUND
* REDIRECT_TYPE_PRODUCT_MOVED_PERMANENTLY
* REDIRECT_TYPE_CATEGORY_MOVED_PERMANENTLY
* REDIRECT_TYPE_PRODUCT_FOUND
* REDIRECT_TYPE_CATEGORY_FOUND

Reply:
it doesn't shock me do you have any propositions to avoid this misunderstanding? a name with an action verb would be better?


* NOT_FOUND_REDIRECTION
* PERMANENT_PRODUCT_REDIRECTION
* PERMANENT_CATEGORY_REDIRECTION
* TEMPORARY_PRODUCT_REDIRECTION
* TEMPORARY_CATEGORY_REDIRECTION

Reply:
The problem is we can't change the key, and we need to be consistent with what have been done.
They are not exactly http status codes, but they are related to. It's like the variables and other parts of this code said, a redirection type (even if 404 isn't a redirection :/)

Reply:
maybe the 404 one should be removed, and the default behaviour when redirect_type is null would be a 404? but I am afraid the former users won't understand the change

Reply:
done!